### PR TITLE
Update telemetry decoder to handle new data format

### PR DESCRIPTION
**Pull request by mreid-moz**
The decoder should now accept either old- or new-style telemetry
submissions.

### CODE REVIEW

**Comment by mreid-moz** [comment]
Hang on - I forgot to add the "sampleId" field.

**Comment by mreid-moz** [comment]
Argh, I won't get to sampleId today - please r? as-is and I'll add it tomorrow.

File: heka/sandbox/decoders/extract_telemetry_dimensions.lua [hunk 1]
```diff
@@ -17,38 +17,78 @@ Fields      = { sourceName = "telemetry" }
...
-    local ok, parsed = pcall(cjson.decode, msg.Payload)
+    -- Attempt to parse the payload as JSON
+    local parsed
+    ok, parsed = pcall(cjson.decode, msg.Payload)
     if not ok then return -1, parsed end
 
-    msg.Fields.sourceVersion    = tostring(parsed.ver) or UNK_DIM
+    if parsed.ver then
+        -- Old-style telemetry
+        msg.Fields.sourceVersion    = tostring(parsed.ver) or UNK_DIM
```
Comment (trink) [inline heka/sandbox/decoders/extract_telemetry_dimensions.lua]: Don't need the `or` here or line 65

File: heka/sandbox/decoders/extract_telemetry_dimensions.lua [hunk 2]
```diff
@@ -17,43 +18,96 @@ Fields      = { sourceName = "telemetry" }
...
+            return false, result
+        end
+        return true, result
+    end
+
+    return true, payload
+end
+
+function sample(id, sampleRange)
+    if not id then
```
Comment (trink) [inline heka/sandbox/decoders/extract_telemetry_dimensions.lua]: Don't need this check

**Pull request merged**